src/main/java/seedu/address/ui/ContactBox.java

```diff
@@ -17,18 +23,33 @@ public class ContactBox extends UiPart<Region> {
     private ImageView contactLogo;
 
     @FXML
-    private Label contactLabel;
+    private Hyperlink contactLabel;
 
     /**
      * Initialises a ContactListBox.
      *
-     * @param type The type of contact
-     * @param name The name of the contact
+     * @param contact Contact
      */
-    public ContactBox(String type, String name) {
+    public ContactBox(Contact contact) {
         super(FXML);
 
+        String type = contact.getContactType().toString().toLowerCase();
         contactLogo.setImage(new Image(this.getClass().getResourceAsStream("/images/contact/" + type + ".png")));
-        contactLabel.setText(name);
+        contactLabel.setText(contact.toString());
+
+        Alert a = new Alert(Alert.AlertType.ERROR);
+        contactLabel.setOnAction(e -> {
+            if (Desktop.isDesktopSupported()) {
+                try {
+                    Desktop.getDesktop().browse(new URI(contact.getLink()));
+                } catch (IOException ex) {
+                    a.setContentText("Sorry, browser couldn't be opened.");
```

Comment:
Does IOException only happen when browser cannot be opened? But I think it's also okay to just show this error as user might not care/need to know if any other error happens.

Reply:
Any suggestion? we need to show at least a friendly message.